main/main.cpp

```diff
@@ -4245,6 +4244,14 @@ void Main::cleanup(bool p_force) {
 	if (engine) {
 		memdelete(engine);
 	}
+#ifndef _3D_DISABLED
+	if (physics_server_3d_manager) {
+		memdelete(physics_server_3d_manager);
+	}
+	if (xr_server) {
+		memdelete(xr_server);
+	}
+#endif // _3D_DISABLED
```

Comment:
Be careful there, the order of de-init matters. Some bad stuff may happen when running the constructor for those after `globals` and `engine` have been destroyed already.